alpha/template/composite/composite.go

```diff
@@ -40,11 +144,119 @@ func (t *Template) Render(ctx context.Context, config *CompositeConfig, validate
 			}
 		} else {
 			allowedComponents := []string{}
-			for k := range t.CatalogBuilders {
+			for k := range *catalogBuilderMap {
 				allowedComponents = append(allowedComponents, k)
 			}
 			return fmt.Errorf("building component %q: component does not exist in the catalog configuration. Available components are: %s", component.Name, allowedComponents)
 		}
 	}
 	return nil
 }
+
+func (t *Template) builderForSchema(schema string, builderCfg BuilderConfig) (Builder, error) {
+	builderFunc, ok := t.registeredBuilders[schema]
+	if !ok {
+		return nil, fmt.Errorf("unknown schema %q", schema)
+	}
+
+	return builderFunc(builderCfg), nil
+}
+
+func (t *Template) parseCatalogsSpec() (*CatalogConfig, error) {
+
+	// get catalog configurations
+	catalogConfig := &CatalogConfig{}
+	catalogDoc := json.RawMessage{}
+	catalogDecoder := yaml.NewYAMLOrJSONDecoder(t.catalogFile, 4096)
+	err := catalogDecoder.Decode(&catalogDoc)
+	if err != nil {
+		return nil, fmt.Errorf("decoding catalog config: %v", err)
+	}
+	err = json.Unmarshal(catalogDoc, catalogConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling catalog config: %v", err)
+	}
+
+	if catalogConfig.Schema != CatalogSchema {
+		return nil, fmt.Errorf("catalog configuration file has unknown schema, should be %q", CatalogSchema)
+	}
+
+	return catalogConfig, nil
+}
+
+func (t *Template) parseContributionSpec() (*CompositeConfig, error) {
+
+	// parse data to composite config
+	compositeConfig := &CompositeConfig{}
+	compositeDoc := json.RawMessage{}
+	compositeDecoder := yaml.NewYAMLOrJSONDecoder(t.contributionFile, 4096)
+	err := compositeDecoder.Decode(&compositeDoc)
+	if err != nil {
+		return nil, fmt.Errorf("decoding composite config: %v", err)
+	}
+	err = json.Unmarshal(compositeDoc, compositeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling composite config: %v", err)
+	}
+
+	if compositeConfig.Schema != CompositeSchema {
+		return nil, fmt.Errorf("composite configuration file has unknown schema, should be %q", CompositeSchema)
+	}
+
+	return compositeConfig, nil
+}
+
+func (t *Template) newCatalogBuilderMap(catalogs []Catalog, outputType string) (*CatalogBuilderMap, error) {
+
+	catalogBuilderMap := make(CatalogBuilderMap)
+
+	// setup the builders for each catalog
+	setupFailed := false
+	setupErrors := map[string][]string{}
+	for _, catalog := range catalogs {
+		errs := []string{}
+		if catalog.Destination.BaseImage == "" {
```

Comment:
Could be done in a follow up, but is the `.Destination.BaseImage` field still necessary since we aren't running `opm` images anymore? If the image isn't going to be taken into account by the rendering process we should probably remove it (or at least comment it out if we plan to use it later) to prevent confusion

Reply:
I agree that this should be commented-out, since I don't think discussion is dead on it right now, but we definitely don't want to enforce it.  
However, because dependencies are starting to stack up against this PR refactor, I'd prefer to handle in another PR.